src/package1/package1Version.ts

```diff
@@ -0,0 +1,146 @@
+      .retrieve(id)) as unknown as PackagingSObjects.PackageUploadRequest;
+  }
+
+  /**
+   * Executes server-side logic for the package1:display command
+   *
+   * @param connection Connection to the org
+   * @param id: id of the MetadataPackageVersion sObject (starts with 04t)
+   */
+  public static async display(connection: Connection, id: string): Promise<Package1Display[]> {
+    if (!id.startsWith('04t')) {
+      throw messages.createError('invalid04tId', [id]);
+    }
+    const query = `SELECT Id,MetadataPackageId,Name,ReleaseState,MajorVersion,MinorVersion,PatchVersion,BuildNumber FROM MetadataPackageVersion WHERE id = '${id}'`;
+    const results = (await connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query)).records;
+    return results.map((result) => ({
+      MetadataPackageVersionId: result.Id,
+      MetadataPackageId: result.MetadataPackageId,
+      Name: result.Name,
+      ReleaseState: result.ReleaseState,
+      Version: `${result.MajorVersion}.${result.MinorVersion}.${result.PatchVersion}`,
+      BuildNumber: result.BuildNumber,
+    }));
+  }
+
+  /**
+   * Lists package versions available in the org. If package ID is supplied, only list versions of that package,
+   * otherwise, list all package versions
+   *
+   * @param connection Connection to the org
+   * @param id: optional, if present, ID of package to list versions for (starts with 033)
+   * @returns Array of package version results
+   */
+  public static async list(connection: Connection, id?: string): Promise<Package1Display[]> {
+    if (id && !id?.startsWith('033')) {
+      // we have to check that it is present, and starts with 033
+      // otherwise, undefined doesn't start with 033 and will trigger this error, when it shouldn't
+      throw messages.createError('invalid033Id', [id]);
+    }
+    const query = `SELECT Id,MetadataPackageId,Name,ReleaseState,MajorVersion,MinorVersion,PatchVersion,BuildNumber FROM MetadataPackageVersion ${
+      id ? `WHERE MetadataPackageId = '${id}'` : ''
+    } ORDER BY MetadataPackageId, MajorVersion, MinorVersion, PatchVersion, BuildNumber`;
+
+    const queryResult = await connection.tooling.query<PackagingSObjects.MetadataPackageVersion>(query);
+    return queryResult.records?.map((record) => ({
+      MetadataPackageVersionId: record.Id,
+      MetadataPackageId: record.MetadataPackageId,
+      Name: record.Name,
+      ReleaseState: record.ReleaseState,
+      Version: `${record.MajorVersion}.${record.MinorVersion}.${record.PatchVersion}`,
+      BuildNumber: record.BuildNumber,
+    }));
+  }
+
+  /**
+   * Will create a PackageUploadRequest object based on the options provided, will poll for completion if pollingOptions are provided
+   *
+   * @param options: Package1VersionCreateRequest options for the new PackageUploadRequest to be created with
+   * @param pollingOptions: options to set frequency, and duration of polling. Default to not poll
+   */
+  public async create(
```

Comment:
Should probably be static. This function is creating a new 04t, which is what package1Version is supposed to represent.